### PR TITLE
Hooks Conversion

### LINKED ISSUE #9: Convert to React Hooks Version

### PR DESCRIPTION
**Pull request by a-saran**
closes #9

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,104 +1,105 @@
...
+  );
 
-  render() {
-    const { imageLoaded, isValidSrc } = this.state;
-    const { src, alt, objectFit, imageProps, wrapperProps } = this.props;
+  const handleError = useCallback(
+    (errorEvent = null) => {
+      setIsValidSrc(false);
+      if (errorEvent !== null && typeof onError === "function")
+        onError(errorEvent);
```
Comment (KRRISH96) [inline src/index.js]: Please use a block for `{}` for multiline if statement.

File: src/index.js [hunk 2]
```diff
@@ -1,104 +1,105 @@
...
-        {isValidSrc && !imageLoaded && (
-          <div className="smooth-preloader">
-            <span className="loader" />
-          </div>
-        )}
-      </div>
-    );
-  }
-}
+  let imageClassnames = `smooth-image img-${
```
Comment (KRRISH96) [inline src/index.js]: This can be a `const`. I suppose the value is not changed anywhere.

**Review by KRRISH96 — COMMENTED** [review]
Looks Good, Couple of minor changes required before merging.

File: src/index.js (hunk 1 above)
Comment (a-saran) [reply src/index.js]: Resolved in https://github.com/KRRISH96/render-smooth-image-react/pull/10/commits/bfbcd7777d2b9560a2ef216666fa537d2aeedd9b

File: src/index.js (hunk 2 above)
Comment (a-saran) [reply src/index.js]: Resolved in https://github.com/KRRISH96/render-smooth-image-react/pull/10/commits/bfbcd7777d2b9560a2ef216666fa537d2aeedd9b

**Review by KRRISH96 — APPROVED** [review]

**Pull request merged**